gen2-seven-segment-recognition/main.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+import cv2
+import depthai as dai
+import numpy as np
+import argparse
+from depthai_sdk import FPSHandler
+from scipy.special import softmax
+from utils import BeamSearchDecoder
+
+
+# --------------- Arguments ---------------
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-nn', '--nn_path', type=str, help="select model blob path for inference", default=None)
+
+args = parser.parse_args()
+
+NN_PATH = args.nn_path
+
+NN_WIDTH = 750
+NN_HEIGHT = 256
+        
+# --------------- Pipeline ---------------
+#   MonoCamera => Crop => Resize => NN
+# ----------------------------------------
+
+pipeline = dai.Pipeline()
+pipeline.setOpenVINOVersion(version=dai.OpenVINO.VERSION_2021_4)
+
+# MonoCamera
+monoRight = pipeline.create(dai.node.MonoCamera)
+monoRight.setBoardSocket(dai.CameraBoardSocket.RIGHT)
+monoRight.setResolution(dai.MonoCameraProperties.SensorResolution.THE_720_P)
+
+# Crop manip
+manip1 = pipeline.create(dai.node.ImageManip)
+manip1.initialConfig.setCropRect(0, 0, 0.694444444, 0.355555556)
+monoRight.out.link(manip1.inputImage)
```

Comment:
Why the cropping?

Reply:
@Erol444  That's the ROI as the model was trained on 256x750 with the 7 seg display filling the majority of it, resizing the whole frame will change the aspect ratio it was trained on and the 7 seg display will be a small part of it
This is recognition only without detection

Reply:
@N950 so to preserve aspect ratio, wouldn't it be more efficient to just set the `resizeManip.setKeepAspectRatio(True)`, instead of having a cropping manip?

Reply:
@Erol444 Aspect ratio and also ROI
`resizeManip.setKeepAspectRatio(True)` would still take a larger part of the frame
Here is what I used for reference on this

Left to right: full frame | setKeepAspectRatio | crop + resize
![aspect_ratio](https://user-images.githubusercontent.com/47245914/199172613-c01e6887-7dda-4393-ab7c-e9fed5e51851.jpg)